src/cargo/util/toml/mod.rs

```diff
@@ -1342,7 +1342,22 @@ impl TomlManifest {
             .unwrap()
             .clone();
         package.workspace = None;
-        package.resolver = Some(ws.resolve_behavior().to_manifest());
+        let current_resolver = package
+            .resolver
+            .as_ref()
+            .map(|r| ResolveBehavior::from_manifest(r))
+            .unwrap_or_else(|| {
+                package
+                    .edition
+                    .as_ref()
+                    .and_then(|e| e.as_defined())
+                    .map(|e| Edition::from_str(e))
+                    .unwrap_or(Ok(Edition::Edition2015))
+                    .map(|e| e.default_resolve_behavior())
+            })?;
+        if ws.resolve_behavior() != current_resolver {
+            package.resolver = Some(ws.resolve_behavior().to_manifest());
+        }
```

Comment:
Can you add a comment explaining why this is needed?

Reply:
Can you expand on the comment about avoiding inserting `resolver` for old `rust-version`s?